library/src/blas3/rocblas_syrk.hpp

```diff
@@ -0,0 +1,369 @@
+          rocblas_int DIM_XYT,
+          typename TScal,
+          typename TConstPtr,
+          typename TPtr>
+__global__ void syrk_herk_kernel(bool              upper,
+                                 rocblas_operation transA,
+                                 rocblas_int       n,
+                                 rocblas_int       k,
+                                 TScal             alpha_host_device,
+                                 TConstPtr         AP_array,
+                                 ptrdiff_t         shift_a,
+                                 rocblas_int       lda,
+                                 rocblas_stride    stride_a,
+                                 TPtr              CP_array,
+                                 ptrdiff_t         shift_c,
+                                 rocblas_int       ldc,
+                                 rocblas_stride    stride_c)
+{
+
+    auto A     = load_ptr_batch(AP_array, hipBlockIdx_z, shift_a, stride_a);
+    auto C     = load_ptr_batch(CP_array, hipBlockIdx_z, shift_c, stride_c);
+    auto alpha = load_scalar(alpha_host_device);
+
+    // compute A^T * A or A * A^T and accumulate on the fly into C
+    // when HERM does A^H in place of A^T
+
+    if(alpha == 0)
+        return;
+    syrk_herk_mult_add_device<HERM, TRANS, DIM_XYT>(upper, n, k, alpha, A, lda, C, ldc);
+}
+
+template <typename TScal, typename TConstPtr, typename TPtr>
+rocblas_status rocblas_syrk_arg_check(rocblas_handle    handle,
+                                      rocblas_fill      uplo,
+                                      rocblas_operation transA,
+                                      rocblas_int       n,
+                                      rocblas_int       k,
+                                      TScal             alpha,
+                                      TConstPtr         AP,
+                                      rocblas_int       offsetA,
+                                      rocblas_int       lda,
+                                      rocblas_stride    strideA,
+                                      TScal             beta,
+                                      TPtr              CP,
+                                      rocblas_int       offsetC,
+                                      rocblas_int       ldc,
+                                      rocblas_stride    strideC,
+                                      rocblas_int       batch_count)
+{
+    if(uplo != rocblas_fill_lower && uplo != rocblas_fill_upper)
+        return rocblas_status_invalid_value;
+    if(transA != rocblas_operation_none && transA != rocblas_operation_transpose)
+        return rocblas_status_invalid_value;
+
+    if(n < 0 || k < 0 || batch_count < 0 || ldc < n || (transA == rocblas_operation_none && lda < n)
+       || (transA != rocblas_operation_none && lda < k))
+        return rocblas_status_invalid_size;
+    if(!n || !batch_count)
+        return rocblas_status_success;
+    if(!AP || !CP || !alpha || !beta)
```

Comment:
Same as comment in herk.

Reply:
Use the Fortran philosophy: If an argument is not required for the evaluation of a function, then do not reference it or care whether it is valid.